jsonschema-core.xml

```diff
@@ -1830,6 +1820,74 @@
+                        within the referreing document.
+                    </t>
+                    <t>
+                        Each embedded JSON Schema Resource (and the document's root) MUST identify itself
+                        with an absolute URI using the "$id" keyword, and SHOULD make use of the "$schema"
+                        keyword to identify the dialect it is using, in the root of the schema resource.
+                    </t>
+                    <t>
+                        When a by-reference applicator is dereferenced as part of the bundling process, the Schema Resource
+                        MUST NOT be embedded by replacing the schema object from which it was referenced, or by wrapping
+                        in other applicator keywords. Instead, a bundled Schema Resource MUST be located as a value of
+                        a "$defs" object at the containing schemas root. The key of the "$defs" for the now embedded
+                        Schema Resource MAY be the "$id" of the bundled schema or some other form of application defined
+                        unique identifer (such as a UUID). This key is not intended to be referenced in JSON Schema, but may
+                        be used by an application to aid the bundling process.
+                    </t>
+                    <t>
+                        References in the containing schema document to the previously external Schema Resources
+                        MUST NOT be changed, and now resolve to a schema using the "$id" of an embedded Schema Resource.
+                    </t>
+                </section>
+                <section title="Differing and Default Dialects">
+                    <t>
+                        In addition to the rules defining the use of
+                        <xref target="keyword-schema">the "$schema" keyword</xref>,
+                        if multiple schema resources are present in a single document, then
+                        schema resources which do not have a "$schema" keyword in their root
+                        schema object MUST be processed as if "$schema" were present with the
+                        same value as for the immediately enclosing resource.
+                    </t>
+                    <t>
+                        Embedded schema resources MAY specify different "$schema" values from their
+                        enclosing resource, as any schema that can be referenced can also be embedded.
+                    </t>
+                </section>
+                <section title="Validating">
+                    <t>
+                        Given a Compound Schema Document may have embedded resources which idenfiy as using different
+                        dialects, it is NOT RECOMMENDED that these documents are validated by applying a meta-schema
+                        to the Compound Schema Document as an instance. It is RECOMMENDED that an alternate
```

Comment:
Is this actually a MUST/MUST NOT?  Treating the compound schema document as an instance is just flat-out incorrect and won't work in the general case.

I think the simplest way to state the requirement here is that each Schema Resource must be separately validated against its meta-schema, _unless_ all schema resources in the Compound Schema Document have the same value for `$schema` (or omit it).  I'm sure the wording could be better, but I think starting from separate per-resource validation and then (as you so a paragraph down) noting that there is a specific case where you can validate an entire document is the way to go.

Reply:
Yes, this is what I was trying to say without actually saying it, but I guess there's no reason not to specify this.
RECOMMENDED is like SHOULD, in that there may be legitimate situations where you don't want to do that (as you've identified).

I feel it's bad form to say "blah blah MUST do x unless something else and in this situation" because it feels a little bit missleading.

I'm still going to have a stab at rephrasing this to make the intent a lot clearer, as you've articulated the intent better.

Reply:
Actually, I've already said this is the case...

                        A Compound Schema Document in which all embedded resources idenfity as using the same
                        dialect MAY be validated by applying the appropriate meta-schema.

Just I didn't specify `$schema`

Reply:
Ah, I see, I think you typoed. /so/do/ makes sense.

Still, update pending.

Reply:
I believe this has been resolved.